sap/xssec/security_context.py

```diff
@@ -538,15 +553,30 @@ def request_token_for_client(self, service_credentials, scopes=''):
 
         :return: Token.
         """
-        if scopes is None:
-            scopes = ''
-
         _check_if_valid(service_credentials, 'service_credentials')
-        for prop in ['clientid', 'clientsecret', 'url']:
-            if prop not in service_credentials:
-                raise ValueError(
-                    '"{0}" not found in "service_credentials"'.format(prop))
-        return self._get_user_token(service_credentials, scopes)
+
+        mtls_attrs = ("url", "clientid", "certificate")  # key is optional
+        cred_attrs = ("url", "clientid", "clientsecret")
+        use_mtls = all(k in service_credentials for k in mtls_attrs)
+        use_cred = not use_mtls and all(k in service_credentials for k in cred_attrs)
+        if not use_mtls and not use_cred:
+            raise ValueError('Either [{}] or [{}] are required in "service_credentials"'.format(
+                ','.join(mtls_attrs), ','.join(cred_attrs)))
+
+        url = '{}/oauth/token'.format(service_credentials['url'].replace('.authentication.', '.authentication.cert.')
```

Comment:
there was a last minute change.
There is now a "certurl" in the VCAP services config object. So we do not need to the domain on our own.
Please use this.

Reply:
updated